### PR TITLE
fix: Enhance API Key Detail Page: Change Permissions Visualization to Tree Format

### LINKED ISSUE #2195: [🕹️] API Key sub-optimal display permissions of permissions
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

Currently, the way assigned / not assigned permissions of a key are displayed is not the most optimal way I think.

For me, this is the result of a few points: 

1. It's displayed as a graph => I don't really understand why but a checkbox tree would be more accessible and is also used in the Authorization -> Roles -> SomeRole -> Permissions tab which would be more consistent.
2. With a lot of permissions the graph just becomes unreadable as its too stacked vertically 
3. What is the point of having the ability to drag permissions onto a key if it does nothing
4. The permission names are cut off and there is no tooltip when hovering over them showing the full name

 I think the way assigned / not assigned permissions of a key are displayed is not the most optimal way

### Steps to Reproduce

1. Create loads of permissions and assign some of them to roles
2. Assign a single role to the key 
3. See how it looks
Please see attached video

### Expected behavior

I would expect this to be shown more straightforwardly, maybe just by taking the checkbox tree instead of this graph

### Other information

https://github.com/user-attachments/assets/6c6de094-c0c5-4b12-9877-bc42a5caf302

### Screenshots

_No response_

### Version info

```markdown
-
```

### PR DESCRIPTION
**Pull request by RajuGangitla**
In this one, i changed the way of showing permissions in api key detail page previously we are using chart i changed it to tree structure 

Fixes #2195 

https://github.com/user-attachments/assets/3fc6592e-af23-4fdc-973d-df8d0423b00f

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [Contributing Guide](./CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [ ] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced the `PermissionTree` component for enhanced visualization of roles and permissions.
	- Added the `APIKeyDetailPage` component for improved API key management.
- **Improvements**
	- Enhanced the layout structure and font loading behavior across multiple applications for better performance and user experience.
	- Updated the `RecursivePermission` component for improved usability in permission management.
	- Improved clarity and consistency in type definitions for better developer experience.

### CODE REVIEW

**Comment by chronark** [comment]
please sign the CLA and address the comments above

**Comment by RajuGangitla** [comment]
i address the comments above

**Comment by chronark** [comment]
CleanShot 2024-10-09 at 13 00 30@2x

**Comment by RajuGangitla** [comment]
changed the title

**Comment by chronark** [comment]
Great work!

**Review by chronark — APPROVED** [review]

**Comment by RajuGangitla** [comment]
@chronark do i need to change anything else

**Review by chronark — CHANGES_REQUESTED** [review]
This PR changes font rendering in 4 apps, can we revert that please?
what does this solve? How is it related to this PR?

File: apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/page.tsx [hunk 1]
```diff
@@ -308,7 +342,8 @@ export default async function APIKeyDetailPage(props: {
           </div>
         </div>
 
-        <Chart
+        <PermissionTree roles={roleTee} />
+        {/* <Chart
```
[the remark this answers is not in the document]
Comment (RajuGangitla) [reply apps/dashboard/app/(app)/apis/[apiId]/keys/[keyAuthId]/[keyId]/page.tsx:346]: Give me a minute I will fix it

**Comment by RajuGangitla** [comment]
@chronark can you please check now

**Review by chronark — APPROVED** [review]

**Comment by RajuGangitla** [comment]
finally wait is over

**Pull request merged**